pkg/std/resource.go

```diff
@@ -0,0 +1,54 @@
+package std
+
+import (
+	"crypto/rand"
+	"crypto/sha256"
+	"fmt"
+)
+
+// ModuleResources keeps track of the base paths for modules, as well
+// as generating the magic modules when they are imported.
+type ModuleResources struct {
+	// module hash -> basePath for resource reads
+	modules map[string]string
```

Comment:
I *think* we can't manipulate this map from different goroutines, so we indeed don't need locking?

Reply:
At present it won't happen, but I could be defensive and put a RWLock on it